### PR TITLE
[New Rules] Potential portscan detected

### PR DESCRIPTION
**Pull request by 1337-42**
## Summary
This PR adds new rules for detecting potential port scans based on thresholds. 
- Potential Network Sweep Detected
- Potential Network Scan Detected
- Potential SYN-Based Network Scan Detected

Because the `network_traffic` integration is added the schemas and manifests are also updated. 
The `network_traffic` integration is added to the `NON_DATASET_PACKAGES` definition for compatibility.

cc: @Aegrah 

### Potential Network Sweep Detected
This rule identifies a potential network sweep.  A network sweep is a method used by attackers to scan a target 
network, identifying active hosts, open ports, and available services to gather information on vulnerabilities and 
weaknesses. This reconnaissance helps them plan subsequent attacks and exploit potential entry points for unauthorized 
access, data theft, or other malicious activities. This rule proposes threshold logic to check for connection attempts 
from one source host to 10 or more destination hosts on commonly used network services.

image
image

### Potential Network Scan Detected
This rule identifies a potential port scan. A port scan is a method utilized by attackers to systematically scan a 
target system or network for open ports, allowing them to identify available services and potential vulnerabilities. 
By mapping out the open ports, attackers can gather critical information to plan and execute targeted attacks, gaining 
unauthorized access, compromising security, and potentially leading to data breaches, unauthorized control, or further 
exploitation of the targeted system or network. This rule proposes threshold logic to check for connection attempts 
from one source host to 20 or more destination ports.

image
image

### Potential SYN-Based Network Scan Detected
This rule identifies a potential SYN-Based port scan. A SYN port scan is a technique employed by attackers to scan a 
target network for open ports by sending SYN packets to multiple ports and observing the response. 
Attackers use this method to identify potential entry points or services that may be vulnerable to exploitation, 
allowing them to launch targeted attacks or gain unauthorized access to the system or network, compromising its 
security and potentially leading to data breaches or further malicious activities. This rule proposes threshold logic 
to check for connection attempts from one source host to 10 or more destination ports using 2 or less packets per port.

image
image

### CODE REVIEW

**Review by eric-forte-elastic — APPROVED** [review]
Manual review of the rules, LGTM :+1: Thanks for the great rule descriptions @1337-42!

File: rules/network/discovery_potential_network_sweep_detected.toml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+from = "now-9m"
+index = ["logs-endpoint.events.network-*", "logs-network_traffic.*", "packetbeat-*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "Potential Network Sweep Detected"
+
+risk_score = 47
+rule_id = "781f8746-2180-4691-890c-4c96d11ca91d"
+severity = "medium"
+tags = ["Elastic", "Host", "Threat Detection", "Discovery", "Network", "Reconnaissance"]
```
Comment (imays11) [inline rules/network/discovery_potential_network_sweep_detected.toml:27]:
```suggestion
tags = ["Domain: Network", "Tactic: Discovery", "Tactic: Reconnaissance", "Use Case: Network Security Monitoring"]
```

File: rules/network/discovery_potential_port_scan_detected.toml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+from = "now-9m"
+index = ["logs-endpoint.events.network-*", "logs-network_traffic.*", "packetbeat-*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "Potential Network Scan Detected"
+
+risk_score = 47
+rule_id = "0171f283-ade7-4f87-9521-ac346c68cc9b"
+severity = "medium"
+tags = ["Elastic", "Host", "Threat Detection", "Discovery", "Network", "Reconnaissance"]
```
Comment (imays11) [inline rules/network/discovery_potential_port_scan_detected.toml:28]:
```suggestion
tags = ["Domain: Network", "Tactic: Discovery", "Tactic: Reconnaissance", "Use Case: Network Security Monitoring"]
```

File: rules/network/discovery_potential_syn_port_scan_detected.toml [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+from = "now-9m"
+index = ["logs-endpoint.events.network-*", "logs-network_traffic.*", "packetbeat-*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "Potential SYN-Based Network Scan Detected"
+
+risk_score = 47
+rule_id = "bbaa96b9-f36c-4898-ace2-581acb00a409"
+severity = "medium"
+tags = ["Elastic", "Host", "Threat Detection", "Discovery", "Network", "Reconnaissance"]
```
Comment (imays11) [inline rules/network/discovery_potential_syn_port_scan_detected.toml:28]:
```suggestion
tags = ["Domain: Network", "Tactic: Discovery", "Tactic: Reconnaissance", "Use Case: Network Security Monitoring"]
```

**Review by imays11 — APPROVED** [review]
Nice set of rules! I updated the tags based on our new tags criteria

**Comment by eric-forte-elastic** [comment]
You may need to copy over the `integration-manifests.json.gz` and `integration-schemas.json.gz` from main. I think there have been some updates that you may need.

**Comment by terrancedejesus** [comment]
@1337-42 @Aegrah 

I copied the integration manifests and schemas from main into this PR, so there is no diff issue now. Aside from this, since `network_traffic` is set in the `NON_DATASET_PACKAGES` variable in `definitions.py` there is no need to add `event.dataset` to the query. `network_traffic` package has several datastreams as you may be aware but these are not separate integrations. Since the fields your queries have specified are in ECS alone, there is no need to specify a specific `network_traffic` data stream to validate fields on. 

PR should be good to go 👍🏽

**Review by terrancedejesus — APPROVED** [review]
Good to go. Thanks for the additional network rules!

**Comment by 1337-42** [comment]
Thanks @Aegrah @eric-forte-elastic @imays11 @terrancedejesus  for the reviews and additional work!

**Pull request merged**